projects/schema-form/src/lib/model/formproperty.ts

```diff
@@ -291,11 +307,184 @@ export abstract class FormProperty {
+    formProperty._propertyBindingRegistry.getPropertyBindingsVisibility().add(dependencyPath, formProperty.path);
+  }
+
+
+  /**
+   * Finds all <code>formProperties</code> from a path with wildcards.<br/>
+   * e.g: <code>/garage/cars/&#42;/tires/&#42;/name</code><br/>
+   * @param target
+   * @param propertyPath
+   */
+  findProperties(target: FormProperty, propertyPath: string): FormProperty[] {
+    const props: FormProperty[] = [];
+    const paths = this.findPropertyPaths(target, propertyPath);
+    for (const path of paths) {
+      const p: FormProperty = target.searchProperty(path);
+      if (p) {
+        props.push(p);
+      }
+    }
+    return props;
+  }
+
+  /**
+   * Creates canonical paths from a path with wildcards.
+   * e.g:<br/>
+   * From:<br/>
+   * <code>/garage/cars/&#42;/tires/&#42;/name</code><br/>
+   * it creates:<br/>
+   * <code>/garage/cars/0/tires/0/name</code><br/>
+   * <code>/garage/cars/0/tires/1/name</code><br/>
+   * <code>/garage/cars/0/tires/2/name</code><br/>
+   * <code>/garage/cars/0/tires/3/name</code><br/>
+   * <code>/garage/cars/1/tires/0/name</code><br/>
+   * <code>/garage/cars/2/tires/1/name</code><br/>
+   * <code>/garage/cars/3/tires/2/name</code><br/>
+   * <code>/garage/cars/3/tires/3/name</code><br/>
+   * <code>/garage/cars/&#42;/tires/&#42;/name</code><br/>
+   * <code>/garage/cars/&#42;/tires/2/name</code><br/>
+   * <code>/garage/cars/&#42;/tires/3/name</code><br/>
+   * <br/>etc...
+   * @param target
+   * @param path
+   * @param parentPath
+   */
+  findPropertyPaths(target: FormProperty, path: string, parentPath?: string): string[] {
+    const ix = path.indexOf('*');
+    if (-1 !== ix) {
+      const prePath = ix > -1 ? path.substring(0, ix - 1) : path;
+      const subPath = ix > -1 ? path.substring(ix + 1) : path;
+      const prop: FormProperty = target.searchProperty(prePath);
+      let pathFound = [];
+      if (prop instanceof PropertyGroup) {
+        const arrProp = prop.properties as FormProperty[];
+        for (let i = 0; i < arrProp.length; i++) {
+          const curreItemPath = (parentPath || '') + prePath + (prePath.endsWith('/') ? '' : '/') + i + subPath;
+          const curreItemPrePath = (parentPath || '') + prePath + i;
+          if (-1 === curreItemPath.indexOf('*')) {
+            pathFound.push(curreItemPath);
+          }
+          const childrenPathFound = this.findPropertyPaths(arrProp[i], subPath, curreItemPrePath);
+          pathFound = pathFound.concat(childrenPathFound);
+        }
+      }
+      return pathFound;
+    }
+    return [path];
+  }
 }
 
 export abstract class PropertyGroup extends FormProperty {
 
-  properties: FormProperty[] | { [key: string]: FormProperty } = null;
+  _properties: FormProperty[] | { [key: string]: FormProperty } = null;
+
+  get properties() {
+    return this._properties;
+  }
+
+  set properties(properties: FormProperty[] | { [key: string]: FormProperty }) {
```

Comment:
I think this method is too big, it makes it pretty difficult to understand. Maybe you could extract few sub methods like `checkCanonicalPath`, `rebindVisibility`, etc. ?

Reply:
done, thx